pkg/internal/flock/flock_unix.go

```diff
@@ -30,6 +38,9 @@ func Acquire(path string) error {
 
 	// We don't need to close the fd since we should hold
 	// it until the process exits.
-
-	return unix.Flock(fd, unix.LOCK_EX)
+	err = unix.Flock(fd, unix.LOCK_NB|unix.LOCK_EX)
+	if errors.Is(err, unix.EWOULDBLOCK) { // This condition requires LOCK_NB.
+		return errors.Wrapf(ErrAlreadyLocked, "cannot lock file %q", path)
```

Comment:
Nit, can we just use the stdlib wrapping (`fmt.Errorf("cannot lock file %q: %w", path, err)`?

Reply:
This wouldn't wrap things properly, and we can't use `errors.Is` on the other side

Reply:
Err what? If you wrap using `fmt.Errorf("%w", err)`, `errors.Is` should absolutely work

Reply:
Ah ok, I didn't know that was the case, fixed.